osu.Game.Rulesets.Mania/Mods/ManiaModScrollUp.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Localisation;
+using osu.Game.Configuration;
+using osu.Game.Overlays.Settings;
+using osu.Game.Rulesets.Mania.Objects;
+using osu.Game.Rulesets.Mania.UI;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Rulesets.UI;
+using osu.Game.Scoring;
+using osu.Game.Screens.Play;
+using static osu.Game.Rulesets.Mania.ManiaSettingsSubsection;
+
+namespace osu.Game.Rulesets.Mania.Mods
+{
+    public class ManiaModScrollUp : Mod, IApplicableToDrawableRuleset<ManiaHitObject>, IApplicableToScoreProcessor, IApplicableToPlayer
+    {
+        public override string Name => "Scroll up";
+        public override string Acronym => "SU";
+        public override LocalisableString Description => @"Key will become faster..., until you miss";
+        public override double ScoreMultiplier => 1;
+        public override IconUsage? Icon => FontAwesome.Solid.AngleDoubleUp;
+        public override ModType Type => ModType.Conversion;
+
+        private DrawableManiaRuleset drawableRuleset = null!;
+
+        private readonly BindableDouble scrollTime = new BindableDouble();
+
+        [SettingSource("Max Speed combo", "The combo count at which point the spead reaches max.")]
+        public BindableInt MaxComboCount { get; } = new BindableInt(480)
+        {
+            MinValue = 1,
+            MaxValue = 2000,
+        };
+
+        [SettingSource("Min Speed", "The speed when combo is 0", SettingControlType = typeof(SettingsSlider<double, ManiaScrollSlider>))]
+        public BindableDouble MinScoreSpeed { get; } = new BindableDouble(DrawableManiaRuleset.MAX_TIME_RANGE)
+        {
+            MinValue = DrawableManiaRuleset.MIN_TIME_RANGE,
+            MaxValue = DrawableManiaRuleset.MAX_TIME_RANGE,
+            Precision = 5,
+        };
+
+        [SettingSource("Max Speed", "The Max speed will reach", SettingControlType = typeof(SettingsSlider<double, ManiaScrollSlider>))]
+        public BindableDouble MaxScoreSpeed { get; } = new BindableDouble(DrawableManiaRuleset.MIN_TIME_RANGE)
+        {
+            MinValue = DrawableManiaRuleset.MIN_TIME_RANGE,
+            MaxValue = DrawableManiaRuleset.MAX_TIME_RANGE,
+            Precision = 5,
+        };
+
+        public ManiaModScrollUp()
+        {
+            MinScoreSpeed.BindValueChanged(s =>
+            {
+                MaxScoreSpeed.MaxValue = s.NewValue;
+            });
+
+            MaxScoreSpeed.BindValueChanged(s =>
+            {
+                MinScoreSpeed.MinValue = s.NewValue;
+            });
```

Comment:
so touching either value sets the other to the same value too? very weird. compare `ModWind{Up,Down}` for how this should actually be done